kong/plugins/log-serializers/runscope.lua

```diff
@@ -0,0 +1,29 @@
+local _M = {}
+
+function _M.serialize(ngx)
+  
+  -- JSON format below based on Runscope Messages resource:
+  -- https://www.runscope.com/docs/api/messages#message-create
+  -- Not sent: req/res:body_encoding, req:form, res:reason
+  return {
+    request = {
+      url = ngx.var.scheme.."://"..ngx.var.host..":"..ngx.var.server_port..ngx.var.request_uri,
+      method = ngx.req.get_method(),
+      headers = ngx.req.get_headers(),
+      body = ngx.ctx.runscope.req_body
+    },
+    response = {
+      status = ngx.status,
+      headers = ngx.resp.get_headers(),
+      size = ngx.var.bytes_sent,
+      body = ngx.ctx.runscope.res_body,
+      timestamp = ngx.req_start_time,
+      response_time = ((ngx.ctx.KONG_ACCESS_TIME or 0) +
+                       (ngx.ctx.KONG_RECEIVE_TIME or 0) +
+                       (ngx.var.upstream_response_time * 1000) +
+                       (ngx.var.request_time * 1000)) / 1000
```

Comment:
I think this part is wrong.

- `ngx.ctx.KONG_ACCESS_TIME`: time for Kong to run the `access_by_lua` directive, akka the time it took Kong before the request was sent to upstream, in `ms`.
- `ngx.ctx.KONG_RECEIVE_TIME`: time spent waiting for the upstream response headers by Kong, in `ms`.
- `ngx.var.upstream_response_time`: same as `KONG_RECEIVE_TIME`, except that it can contain a comma separated list of values if the request was retried by nginx_proxy (that can happen under load, see the `proxy_next_*` directives of nginx_proxy).
- `ngx.var.request_time: `time elapsed between the first bytes were read from the client and the log write after the last bytes were sent to the client`

So this value will actually be 2 to 3 times the actual processing time:

```
ngx.ctx.KONG_ACCESS_TIME  |  |-----------|                                |
ngx.ctx.KONG_RECEIVE_TIME |                     |-------------|           |
$upstream_response_time   |                     |-------------|           |
$request_time             |-----------------------------------------------|
```

The `ngx.ctx.*` variable were added to avoid handling the case when `var` variables are comma-separated strings, causing Lua to failing the `+`/`*` etc operators.

Now, I am not sure what the `response_time` value is supposed to represent (time it took for the client (downstream) to get a response from `upstream`, through Kong? If so, the most accurate way is to mimic what the ALF serializer does: [alf.lua#L110](https://github.com/Mashape/kong/blob/next/kong/plugins/log-serializers/alf.lua#L110) by adding the `PROXY_LATENCY` (time taken for Kong to receive and proxy to upstream) + `WAITING_TIME` (time elapsed until Kong receives the response headers) + `RECEIVE_TIME` (time taken for the response body to be proxied back to downstream). Those values are also already in `ms` resolution.

If this is what `response_time` is supposed to reflect, I can make those changes myself if you want.